proseco/fid.py

```diff
@@ -518,4 +536,26 @@ def get_fid_positions(detector, focus_offset, sim_offset):
 
     yang, zang = np.degrees(yfid) * 3600, np.degrees(zfid) * 3600
 
+    enable_fid_offset_env = os.environ.get("PROSECO_ENABLE_FID_OFFSET")
+    if enable_fid_offset_env not in ("True", "False", None):
+        raise ValueError(
+            f'PROSECO_ENABLE_FID_OFFSET env var must be either "True", "False", or None, '
+            f"got {enable_fid_offset_env}"
+        )
+
+    # If enable_fid_offset_env is None or True then try to apply an offset
+    if not enable_fid_offset_env or enable_fid_offset_env == "True":
+        # Require that t_ccd and date are provided if fid offset is enabled
+        if enable_fid_offset_env == "True" and (t_ccd is None or date is None):
+            raise ValueError(
+                "t_ccd_acq and date must be provided if PROSECO_ENABLE_FID_OFFSET is 'True'"
+            )
+
+        # Apply fid offset if t_ccd and date are provided
+        # (this is an "if" the case when enable_fid_offset_env is None)
+        if t_ccd is not None and date is not None:
+            dy, dz = get_fid_offset(date, t_ccd)
+            yang += dy
+            zang += dz
```

Comment:
You can reduce the complexity like so, to the point that code comments are not even needed:
```suggestion
    has_tccd_and_date = t_ccd is not None and date is not None

    if enable_fid_offset_env == "True" and not has_tccd_and_date:
        raise ValueError(
            "t_ccd_acq and date must be provided if PROSECO_ENABLE_FID_OFFSET is 'True'"
        )

    if has_tccd_and_date and enable_fid_offset_env != "False":
        dy, dz = get_fid_offset(date, t_ccd)
        yang += dy
        zang += dz
```